### DISCUSSION TITLE
Is it possible to make a command to run (netwide) assembly code?

### DISCUSSION THREAD

**Discussion by sh0tx420** [Rapptz/discord.py#6026] [Q&A] [answered]
I want to make a command that can run nasm straight from discord but I'm not really sure how. Could anyone help me with this or is that even possible?

**Comment by ShineyDev** [comment, marked as answer]
It's possible.
Abstract this idea from Discord, since you're only collecting input and sending output. This way it's as simple as running a shell command.

- You need to collect the input from Discord, parse it out of code fencing or otherwise.
- Send this input to be compiled by NASM via an [asyncio subprocess](https://docs.python.org/3/library/asyncio-subprocess.html).
- Run your compiled program.
- Collect the output from the process, this process is explained [here](https://docs.python.org/3/library/asyncio-subprocess.html#asyncio.asyncio.subprocess.Process.communicate).
- Send this output back to Discord.

**Reply by sh0tx420** [reply]
thank you
